app/javascript/packages/document-capture/components/review-issues-step.tsx

```diff
@@ -74,51 +81,177 @@ function ReviewIssuesStep({
+  }
 
   // let FormSteps know, via FormStepsContext, whether this page
   // is ready to submit form values
   useEffect(() => {
     changeStepCanComplete(!!hasDismissed);
   }, [hasDismissed]);
 
+  useEffect(() => {
+    if (!inPersonURL || isFailedResult) {
+      return;
+    }
+    if (inPersonCtaVariantActive === 'in_person_variant_a') {
+      trackEvent('IdV: IPP CTA Variant A');
+    } else if (inPersonCtaVariantActive === 'in_person_variant_b') {
+      trackEvent('IdV: IPP CTA Variant B');
+    } else if (inPersonCtaVariantActive === 'in_person_variant_c') {
+      trackEvent('IdV: IPP CTA Variant C');
+    }
+  }, []);
+
   if (!hasDismissed) {
     if (pii) {
       return <BarcodeAttentionWarning onDismiss={onWarningPageDismissed} pii={pii} />;
     }
 
-    return (
-      <Warning
-        heading={t('errors.doc_auth.throttled_heading')}
-        actionText={t('idv.failure.button.warning')}
-        actionOnClick={onWarningPageDismissed}
-        location="doc_auth_review_issues"
-        remainingAttempts={remainingAttempts}
-        troubleshootingOptions={
-          <DocumentCaptureTroubleshootingOptions
-            location="post_submission_warning"
-            showAlternativeProofingOptions={!isFailedResult}
-            heading={t('components.troubleshooting_options.ipp_heading')}
-          />
-        }
-      >
-        {!!unknownFieldErrors &&
-          unknownFieldErrors
-            .filter((error) => !['front', 'back'].includes(error.field!))
-            .map(({ error }) => <p key={error.message}>{error.message}</p>)}
-
-        {remainingAttempts <= DISPLAY_ATTEMPTS && (
-          <p>
-            <strong>{t('idv.failure.attempts', { count: remainingAttempts })}</strong>
-          </p>
-        )}
-      </Warning>
-    );
+    if (!inPersonURL || isFailedResult) {
+      return (
+        <Warning
+          heading={t('errors.doc_auth.throttled_heading')}
+          actionText={t('idv.failure.button.warning')}
+          actionOnClick={onWarningPageDismissed}
+          location="doc_auth_review_issues"
+          remainingAttempts={remainingAttempts}
+          troubleshootingOptions={
+            <DocumentCaptureTroubleshootingOptions
+              location="post_submission_warning"
+              showAlternativeProofingOptions={!isFailedResult}
+              heading={t('components.troubleshooting_options.ipp_heading')}
+            />
+          }
+        >
+          {!!unknownFieldErrors &&
+            unknownFieldErrors
+              .filter((error) => !['front', 'back'].includes(error.field!))
+              .map(({ error }) => <p key={error.message}>{error.message}</p>)}
+
+          {remainingAttempts <= DISPLAY_ATTEMPTS && (
+            <p>
+              <strong>{t('idv.failure.attempts', { count: remainingAttempts })}</strong>
+            </p>
+          )}
+        </Warning>
+      );
+    }
+    if (inPersonCtaVariantActive === 'in_person_variant_a') {
```

Comment:
can we write a separate function that returns the warning + DocumentCaptureTroubleshootingOptions components w/ the variant input? They appear to share a lot of the same values so I'm thinking we can reduce some of the code by placing them in a function where we have params for the bits that are diff. based on which variant the user sees

Reply:
They definitely aren't DRY :) It was decided that a refactor was out of scope for this ticket and that the follow-up work of refactoring would likely not be worthwhile since this code will only be live for about a month. Refactoring is valuable for keeping code maintainable, easy to add to, etc. but this was written with the intent of being easy to remove safely; "don't want to support option B anymore? chop that conditional out".